pageserver/src/layered_repository/timeline.rs

```diff
@@ -278,6 +287,9 @@ pub struct LayeredTimeline {
     wait_lsn_time_histo: Histogram,
     current_physical_size_gauge: UIntGauge,
 
+    /// copy of current_logical_size, for prometheus
+    current_logical_size_gauge: IntGauge,
```

Comment:
Is it necessary to have a copy? For physical size we use prometheus counter directly. It is the same atomic underneath

Reply:
I considered that and was kind of on the fence on which way to go. It seems weird to store an important variable merely in a prometheus gauge. OTOH, it would avoid the race condition.

I wish there was a way to implement a custom prometheus gauge that reads the current value from an existing variable, but I couldn't see an easy way to do that.

Reply:
And do you think the refactoring in https://github.com/neondatabase/neon/pull/2278 makes a difference? I think it makes it look slightly more weird to peek into the prometheus gauge, making the case for having a separate field as in this PR even stronger.

Reply:
Yeah, I agree. It is a bit weird. We can hack something around but I dont think that it will look better. Though maybe worth trying. Having to increment one atomic instead of two and avoiding race condition makes reasonable trade off to me. Though if in some hypothetical case we need to change ordering it will require more changes

> And do you think the refactoring in https://github.com/neondatabase/neon/pull/2278 makes a difference?

Yeah, it was not intended to make a difference. After adding two more metrics my screen overflowed and I decided to split it off so it is easier to see what we keep in timeline struct and how it is built

Reply:
If we want to implement our own metric, there's this trait https://docs.rs/prometheus/latest/prometheus/core/trait.Metric.html

Though I don't think it's worth the effort. Using a Gauge to store the value seems fine.